testgen/generators/datetime_gen.js

```diff
@@ -172,6 +261,29 @@ const dt_fields = {
     'longGeneric': 'vvvv'}
 };
 
+            // Convert to a date for formatting with ECMASript Intl
+function zdt_to_date(zdt, locale, calendar) {
+  // Convert to a date for formatting with ECMASript Intl
+  let date_string;
+  let this_date;
+  try {
+    // Calendar
+    date_string = zdt.toLocaleString(locale, {calendar: calendar});
+  } catch (error) {
+    console.log('@@@@ zdt_to_date: \"%s\" %s, from zdt= %s, locale = %s, cal = %s',
+                error, date_string, zdt, locale, calendar);
+    return null;
+  }
+  try {
+    this_date = new Date(date_string);
+  } catch (error) {
+    console.log('##### new Date: \"%s\"%s, %s from %s', error, date_string, zdt);
+    return null;
+  }
+
+  return this_date;
+}
+
```

Comment:
Please use the Temporal cookbook example on how to perform this conversion. https://tc39.es/proposal-temporal/docs/cookbook.html#temporal-types--legacy-date

`new Date(zdt.epochMilliseconds)`

Reply:
Done